CI/build-mudlet-for-windows.sh

```diff
@@ -0,0 +1,133 @@
+#                   CI/CB system using same variable
+#          1.4.0    Rewrite Makefile to use ccache.exe if available
+#          1.3.0    No changes
+#          1.2.0    No changes
+#          1.1.0    No changes
+#          1.0.0    Original version
+
+# Script to each time to build the Mudlet code currently checked out in
+# ~/src/mudlet in a MINGW32 or MINGW64 shell
+
+# To be used AFTER setup-windows-sdk.sh has been run; once this has completed
+# successfully use:
+# * package-mudlet-for-windows.sh to put everything together in an archive that
+#   can be unzipped and run on a different PC.
+
+# Exit codes:
+# 0 - Everything is fine. 8-)
+# 1 - Failure to change to a directory
+# 2 - Unsupported MSYS2/MINGGW shell type
+# 3 - Unsupported build type
+
+if [ "${MSYSTEM}" = "MSYS" ]; then
+  echo "Please run this script from an MINGW32 or MINGW64 type bash terminal appropriate"
+  echo "to the bitness you want to work on. You may do this once for each of them should"
+  echo "you wish to do both."
+  exit 2
+elif [ "${MSYSTEM}" = "MINGW32" ]; then
+  export BUILD_BITNESS="32"
+  export BUILDCOMPONENT="i686"
+elif [ "${MSYSTEM}" = "MINGW64" ]; then
+  export BUILD_BITNESS="64"
+  export BUILDCOMPONENT="x86_64"
+else
+  echo "This script is not set up to handle systems of type ${MSYSTEM}, only MINGW32 or"
+  echo "MINGW64 are currently supported. Please rerun this in a bash terminal of one"
+  echo "of those two types."
+  exit 2
+fi
+
+MINGW_BASE_DIR="${GHCUP_MSYS2}\mingw32"
+export MINGW_BASE_DIR
+MINGW_INTERNAL_BASE_DIR="/mingw${BUILD_BITNESS}"
+export MINGW_INTERNAL_BASE_DIR
+PATH="${MINGW_INTERNAL_BASE_DIR}/usr/local/bin:${MINGW_INTERNAL_BASE_DIR}/bin:/usr/bin:${PATH}"
+export PATH
+
+echo "MSYSTEM is: ${MSYSTEM}"
+echo "PATH is now:"
+echo "${PATH}"
+echo ""
+
+cd $GITHUB_WORKSPACE || exit 1
+mkdir -p "build-${MSYSTEM}"
+
+cd ${GITHUB_WORKSPACE}/build-"${MSYSTEM}" || exit 1
+
+#### Qt Creator note ####
+# If one is planning to use qtcreator these will probably be wanted in a
+# shell startup script so as to prepare it to use Lua 5.1 when running
+# qmake (needed to process translation files to get the translations
+# statistics):
+LUA_PATH=$(cygpath -u "$(luarocks --lua-version 5.1 path --lr-path)" )
+export LUA_PATH
+LUA_CPATH=$(cygpath -u "$(luarocks --lua-version 5.1 path --lr-cpath)" )
+export LUA_CPATH
+
+echo ""
+echo "Adjusting LUA paths for Lua 5.1:"
+echo "LUA_PATH is: ${LUA_PATH}"
+echo "LUA_CPATH is: ${LUA_CPATH}"
+echo ""
+
+#### Qt Creator note ####
+# The following WITH_XXXXs can usefully be used in the Qt Creator's "Project"
+# tab for the "Kit" concerned in the "Build Environment" section:
+if [ "${MSYSTEM}" = "MINGW64" ]; then
+  # The MINGW64 case already has the Qt5 keychain package pre-built so no need
+  # to build our bundled copy:
+  export WITH_OWN_QTKEYCHAIN="NO"
+fi
+
+# The updater is not helpful in this (build it yourself) environment
+export WITH_UPDATER="NO"
```

Comment:
Yeah definitely.

Reply:
But fix https://github.com/Mudlet/Mudlet/pull/7217#pullrequestreview-2041248681 first - or things will get messy/borked if we start injecting 64-Bit updates into those users who are actually using 32-bit Windows - fair enough it won't be an issue for release builds until Vadi and the rest of us Core Dev team get our act together and get one out but PTBs will adopt this within 24 hours of it being merged into the `development` branch code. Note, of course, that we will have to go through the PTB and release execution paths with a fine-tooth comb because ***they are hard, if not, impossible, to test completely before being used for* real!**